src/puppeteer.js

```diff
@@ -58,7 +69,8 @@ export const launchPuppeteer = (opts = {}) => {
     opts.args = opts.args || [];
     opts.args.push('--no-sandbox');
 
-    if (opts.headless === undefined || opts.headless === null) opts.headless = !!process.env[ENV_VARS.HEADLESS];
+    if (opts.userAgent) opts.args.push(`--user-agent=${opts.userAgent}`);
```

Comment:
There are spaces in user agent so we may need to wrap user agent in commas

Reply:
I've added unit test for this, it works well